### PR TITLE
feat(presenter): double opt in

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

Defines the route `/optin/@code` which checks the given code and either shows an error with an link to resubscribe or shows a success message and automatically redirects to the main page after 5 secs.

### Issues

- fixes https://github.com/dreammall-earth/dreammall.earth/issues/164

### Todo

Uncovered code
image

### CODE REVIEW

**Review by Mogge — CHANGES_REQUESTED** [review]
```
3:52:16 PM [vike][request(24)] URL /optin/2ae4f265372fb56b doesn't match the route of any of your pages:
┌──────────────┬──────────────────┬─────────────────────────┐
│ ROUTE        │ TYPE             │ DEFINED BY              │
├──────────────┼──────────────────┼─────────────────────────┤
│ /            │ Filesystem Route │ /src/pages/index/       │
│ /datenschutz │ Filesystem Route │ /src/pages/datenschutz/ │
│ /impressum   │ Filesystem Route │ /src/pages/impressum/   │
└──────────────┴──────────────────┴─────────────────────────┘
See https://vike.dev/routing for more information about routing.
```

**Comment by Mogge** [comment]
image

Looks terrifying. In success case as well

File: presenter/src/pages/optin/+Page.vue [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    isError.value = !result?.data?.confirmNewsletter
+  } catch (error) {
+    isError.value = true
+  }
+
+  isLoading.value = !isLoading.value
+
+  if (!isError.value) {
+    setTimeout(() => {
+      navigate('/')
```
Comment (Mogge) [inline presenter/src/pages/optin/+Page.vue:46]: @ulfgebhardt do yo agree that this is better?

**Comment by ulfgebhardt** [comment]
Die unit tests terminieren nicht

File: presenter/src/locales/en.json [hunk 2]
```diff
@@ -199,6 +199,15 @@
       "sitemap": "Sitemap"
     }
   },
+  "optin": {
+    "error": {
+      "link": "Erneut eintragen",
+      "text": "Bei der Bestätigung ist ein Fehler aufgetreten. Bitte trage dich erneut in den Newsletter ein."
+    },
+    "success": {
+      "text": "Du hast dein Newsletterabbonoment erfolgreich bestätigt. Du wirst in Kürze auf die Startseite weitergeleitet."
```
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply presenter/src/locales/en.json:208]: Yes, we will have a complete translation when multi language is available

**Review by Elweyn — APPROVED** [review]

**Pull request merged**